### PR TITLE
Add getting started section to the admin cluster dashboard

### PR DESCRIPTION
**Pull request by jerolimov**
**Fixes**: 
Epic: https://issues.redhat.com/browse/ODC-5013
Story: https://issues.redhat.com/browse/ODC-5680

**Analysis / Root cause**: 
As an admin, I want to see the new getting started section with areas to "Set up your cluster" and other Learning resources on the Admin Overview page.

**Solution Description**: 
This PR is based on #8950 which adds a Getting started section to the Add page in the developer perspective. The first two commits should be reviewed as part of this commit.

This PR here should focus on the other two commits and the changes in the `public` package.

The new Getting started section in the admin perspective adds three content cards with different links:

1. "Set up your cluster" shows some good first steps to configure a cluster like "Add identity providers" and "Configure alert receivers". If an identity provider or the alert receivers are configured, these options disappear. If everything is configured the complete column disappears.
2. "Build with guided documentation" - This column shows up to two quick starts. It also prefers two hardcoded quickstarts (Monitoring your sample app and Quarkus with Helm). It but will automatically hide them when the admin completed them. When all quick started are completed, this column disappears.
3. "Explore new admin features" shows some static links to the API Explorer and a link to start customization.

**Screen shots / Gifs for design review**: 
@openshift/team-devconsole-ux

New getting started section:
image

When the user wants to hide them:
image

When the page loads the first time and when the admin revisits the overview again (in one session):
Peek 2021-05-18 02-41

**Unit test coverage report**: 
image

**Test setup:**
* Open the admin perspective as kube:admin
* Check the links in the new getting started section
* Add an identity provider (click on the link, select Basic Auth and enter a URL like `http://localhost`)
* Configure an alert receiver (click on the link, select Configure for Critical and Default receivers, select Webhook and add an URL like `http://localhost`)

**Browser conformance**: 
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by jerolimov** [comment]
/kind feature

Since the epic acceptance criteria of [ODC-5013](https://issues.redhat.com/browse/ODC-5013) have not changed and the epic has `doc-ack` and `px-ack`, we approve this on behalf of them.
/label px-approved
/label docs-approved

Add hold to wait for an `qe-approved` also if someone adds a lgtm!
/hold

**Comment by sanketpathak** [comment]
Tested the PR through cluster-bot. Everything works as expected
Screenshot from 2021-05-18 14-06-44
Screenshot from 2021-05-18 14-19-50

/label qe-approved

**Comment by jerolimov** [comment]
Remove hold because this PR has now also approvals and just needs a lgtm.
/hold cancel

File: frontend/public/components/dashboard/dashboards-page/cluster-dashboard/getting-started/explore-admin-features-getting-started-card.tsx
Comment (jerolimov) [reply frontend/public/components/dashboard/dashboards-page/cluster-dashboard/getting-started/explore-admin-features-getting-started-card.tsx:15]: Hmmm, without prettier this is a one liner :smile:

**Comment by jerolimov** [comment]
Rebased and fixed an issue that the OAuth resource for the "Add identity providers" link was not fetched successfully.

File: frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx
Comment (christianvogt) [inline frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx:221]: Looks like we only care about the result of `getAlertmanagerConfig` the first time because `loadErrorMsg` is set in the state and a subsequent render can never result in a new error message in the state.

This differs from the original impl which could call `setErrorMsg` on subsequent renders.

**Comment by spadgett** [comment]
Should we consider removing the blue kubeadmin notification with this change? This looks like a more accurate check for IDPs, and I'm not sure the banner is still needed.

**Comment by christianvogt** [comment]
@spadgett i'd prefer to not make the change as part of this PR. but something that can be brought up with UX as I don't know the history of the banner. Furthermore the banner would be nagging all pages whereas the getting started card won't and can be hidden.

File: frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx
Comment (jerolimov) [reply frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx]: I updated my PR now which includes also some small fixes for the form when the `getAlertmanagerConfig` returns an error. I saw this different scenarios and tested them on the master branch and with this PR.

You need to update the alertmanager-main secret to reproduce some of these errors.

(console path `/k8s/ns/openshift-monitoring/secrets/alertmanager-main/yaml`)

1. YAML is in a good state: Works fine before and with this PR.

2. There is no `data.alertmanager.yaml`: The complete Alertmanager shows an error (before and with this PR):

image

3. The YAML contains invalid or incompleted data, like a string or an object without missing receivers, etc.:

secret (with base64 encoded yaml `dafa` - a string):
```
data:
  alertmanager.yaml: ZGFmYQ==
```

secret (with base64 encoded yaml `dafa:` - an object):
```yaml
data:
  alertmanager.yaml: ZGFmYToK
```

Before: Crash because some expected properties are not defined in the config.

After: Because the most code use lodash or some other fallbacks (like ` || {}`) I added these fallbacks to the missing senarios. The form shows now an empty, not configured form.


Finally: The loading and patch error message was just shown to the user. The save button was not disabled. Instead of setting the loadError as state variable, I just show the patch or loading message, and prefers the patch message.

Comment (christianvogt) [reply frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx]: The new changes look much better. 
Handling the two error states separately is also cleaner solution.

**Comment by christianvogt** [comment]
/approve

I just haven't run the latest changes that affect the alert manager config.

Getting started cards worked well and lgtm

File: frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx
Comment (invincibleJai) [reply frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx]: /cc @kyoto

File: frontend/public/components/dashboard/dashboards-page/cluster-dashboard/getting-started/getting-started-section.scss
Comment (invincibleJai) [inline frontend/public/components/dashboard/dashboards-page/cluster-dashboard/getting-started/getting-started-section.scss:1]: nit: can we have as bleow?

```suggestion
.co-dashboard-getting-started-section {
```

Comment (jerolimov) [reply frontend/public/components/dashboard/dashboards-page/cluster-dashboard/getting-started/getting-started-section.scss]: Fixed

See https://github.com/openshift/console/compare/59068d40fba7a271e13b96803de8b5a9ed2c6ae6..fa1c708ab298c706378019e68660e990c6a39d34

**Review by sahil143 — COMMENTED** [review]
@jerolimov I'm seeing the identity provider link on the card. Am I missing something? I'm logged in as kubeadmin

Screenshot 2021-05-24 at 10 59 19 PM

**Comment by spadgett** [comment]
> Furthermore the banner would be nagging all pages whereas the getting started card won't and can be hidden.

I think that's the problem :) It's nagging too much with no way to hide.

This history was simply to guide people to create an IDP after installing the cluster (and hopefully disable the kubeadmin user, but you have to be a little careful with that). We can raise with UX.

cc @beanh66

File: frontend/public/components/dashboard/dashboards-page/cluster-dashboard/getting-started/cluster-setup-getting-started-card.tsx
Comment (jerolimov) [reply frontend/public/components/dashboard/dashboards-page/cluster-dashboard/getting-started/cluster-setup-getting-started-card.tsx:38]: We use this really rare in our code base and the imports looks like this:

```tsx
import { chart_color_green_400 as successColor } from '@patternfly/react-tokens/dist/js/chart_color_green_400';
```

Not a big fan of this but I can do this in a follow up PR for the add page and the admin page.

**Comment by jerolimov** [comment]
> @jerolimov I'm not seeing the identity provider link on the card. Am I missing something? I'm logged in as kubeadmin
> 
> Screenshot 2021-05-24 at 10 59 19 PM

**Comment by jerolimov** [comment]
Ooops :smile:

**Comment by jerolimov** [comment]
> @jerolimov I'm not seeing the identity provider link on the card. Am I missing something? I'm logged in as kubeadmin

@sahil143 You can open the console path `/k8s/cluster/config.openshift.io~v1~OAuth/cluster` manually. Some of our cluster have a configured identity provider already. In this case we don't want to show the link again.

I tested this with a fresh cluster-bot cluster and it works as expected.

**Review by sahil143 — APPROVED** [review]
/lgtm

tested locally works as expected

**Review by invincibleJai — APPROVED** [review]
Verified the changes.

**Comment by kyoto** [comment]
FYI @dtaylor113

**Pull request merged**
